### ISSUE TITLE
if everyone leaves a room there's no way to then get into it to delete its alias so you can reuse the alias

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#3194]

**Comment by richvdh** [comment]
Somewhat related to #1361.

you can delete it as a server admin, obvs. I guess we should extend that ability to people who had rights to remove the alias in the last known state of the room.

**Comment by ara4n** [comment]
vaguely relate to https://github.com/matrix-org/synapse/pull/614

**Comment by thekyriarchy** [comment]
oops, i filed a duplicate at matrix-org/synapse/issues/2100

**Comment by t3chguy** [comment]
Can this not now be done by shift clicking or something wacky like that? I recall some weird easter egg about it mentioned

**Comment by thekyriarchy** [comment]
@t3chguy  I don't think so...

**Comment by t3chguy** [comment]
@ara4n I seem to recall you saying something was shift clickable?

**Comment by thekyriarchy** [comment]
i just found this: https://github.com/vector-im/riot-web/pull/1680

**Comment by turt2live** [comment]
Shift-clicking as a server admin in the room directory *should* remove it.

**Comment by thekyriarchy** [comment]
yeah, it only works as a server admin in the room directory, not room admins and not automatically after everyone leaves

**Comment by cryptix** [comment]
Hmm.. I managed to do this but the room is not in the room directory but I still have the internal ID....

**Comment by ara4n** [comment]
the horrible hacky workaround is to delete the right row from room_aliases table in the DB.  we need a proper solution (hence this bug)

**Comment by richvdh** [comment]
can one hit the [`DELETE /_matrix/client/r0/directory/room/{roomAlias}` api](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#delete-matrix-client-r0-directory-room-roomalias) rather than messing with the DB?

**Comment by cryptix** [comment]
Thanks you two! I used the `DELETE` method and can now reuse the alias.

**Comment by Quenz** [comment]
What if the room goes dead, including the admin of the room, and someone else wants to use the room's alias?

**Comment by Justinizer** [comment]
@Quenz Yeah Sudoquai and me got the same problem.
I think this is still an issue.
Why aliases of deleted rooms aren't deleted too?

**Comment by sudoquai** [comment]
We need a solution for this - when can we expect a solution for this ?

**Comment by tulir** [comment]
This should be handled on the server side by removing all aliases when the last local user leaves the room. It's not like the aliases would work without the server being in the room, so not much point in keeping them.

**Comment by t3chguy** [comment]
Closing as a proposed workaround for what is fundamentally a server misbehaviour - see @tulir's message above

**Comment by ara4n** [comment]
I've reopened https://github.com/matrix-org/synapse/issues/1307 with more info, so that we're tracking this (very valid) issue *somewhere*.  It's all very well that [https://github.com/vector-im/element-web/issues/3194#issuecomment-699692620](the fix has been identified), but if it's not tracked on an open bug it's never going to happen.
